### PR TITLE
[MySql] Add maintenance window to update flexible server cmdlet

### PR DESCRIPTION
**Pull request by DaeunYim**
## Description

Added maintenance window parameter for flexible server update cmdlet. [doc file](https://github.com/DaeunYim/azure-powershell/blob/mysql-maintenance-window/src/MySql/docs/Update-AzMySqlFlexibleServer.md#-maintenancewindow)

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by msJinLei — CHANGES_REQUESTED** [review]

File: src/MySql/custom/Update-AzMySqlFlexibleServer.ps1 [hunk 1]
```diff
@@ -181,6 +188,40 @@ function Update-AzMySqlFlexibleServer {
                 $null = $PSBoundParameters.Remove('StorageAutogrow')
             }
 
+            if ($PSBoundParameters.ContainsKey('MaintenanceWindow')) {
```
Comment (msJinLei) [inline src/MySql/custom/Update-AzMySqlFlexibleServer.ps1:191]: please ensure c# codes is going to handle the case when MaintenanceWindow is absent.

File: src/MySql/custom/Update-AzMySqlFlexibleServer.ps1 [hunk 2]
```diff
@@ -181,6 +188,40 @@ function Update-AzMySqlFlexibleServer {
...
+                }
+                else {
+                    $ParsedWindow = $PSBoundParameters.MaintenanceWindow -split ":"
+                    $DaytoNumber = @{Mon = 1; Tue = 2; Wed = 3; Thur = 4; Fri = 5; Sat = 6; Sun = 0}
+                    
+                    if ($ParsedWindow.Length -ge 1){
+                        $PSBoundParameters.MaintenanceWindowDayOfWeek =  $DaytoNumber[$ParsedWindow[0]]
+                    }
+                    
+                    if ($ParsedWindow.Length -ge 2){
```
[the remark this answers is not in the document]
Comment (DaeunYim) [reply src/MySql/custom/Update-AzMySqlFlexibleServer.ps1:211]: The logic is to add value to each week, hour, and minute. I'm afraid  changing it to elseif wouldn't work here

File: src/MySql/custom/Update-AzMySqlFlexibleServer.ps1 [hunk 3]
```diff
@@ -181,6 +188,40 @@ function Update-AzMySqlFlexibleServer {
...
+                        $PSBoundParameters.MaintenanceWindowStartHour = $ParsedWindow[1]
+                    }
+
+                    if ($ParsedWindow.Length -ge 3){
+                        $PSBoundParameters.MaintenanceWindowStartMinute = $ParsedWindow[2]
```
[the remark this answers is not in the document]
Comment (DaeunYim) [reply src/MySql/custom/Update-AzMySqlFlexibleServer.ps1:216]: Here as well.  Changing it to elseif will make the code to 

`if ($ParsedWindow.Length -ge 1){
    $PSBoundParameters.MaintenanceWindowDayOfWeek =  $DaytoNumber[$ParsedWindow[0]]
}

elseif ($ParsedWindow.Length -ge 2){
   $PSBoundParameters.MaintenanceWindowDayOfWeek =  $DaytoNumber[$ParsedWindow[0]]
    $PSBoundParameters.MaintenanceWindowStartHour = $ParsedWindow[1]
}

if ($ParsedWindow.Length -ge 3){
    $PSBoundParameters.MaintenanceWindowDayOfWeek =  $DaytoNumber[$ParsedWindow[0]]
    $PSBoundParameters.MaintenanceWindowStartHour = $ParsedWindow[1]
    $PSBoundParameters.MaintenanceWindowStartMinute = $ParsedWindow[2]
}`

Is this sth you want to recommend??

File: src/MySql/custom/Update-AzMySqlFlexibleServer.ps1 (hunk 1 above)
Comment (DaeunYim) [reply src/MySql/custom/Update-AzMySqlFlexibleServer.ps1:191]: I saw no problem updating a server with parameters other than MaintenanceWindow.

**Review by msJinLei — APPROVED** [review]

**Pull request merged**